src/rdkafka_ssl.c

```diff
@@ -1298,6 +1367,51 @@ int rd_kafka_ssl_ctx_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
         }
 #endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+        if (rk->rk_conf.ssl.engine_location && !rk->rk_conf.ssl.engine) {
+                /* OpenSSL loads an engine as dynamic id and stores it in internal
+                 * list, as per LIST_ADD command below. If engine already exists
+                 * in internal list, it is supposed to be fetched using engine id. 
+                 */
+                rk->rk_conf.ssl.engine = ENGINE_by_id(rk->rk_conf.ssl.engine_id);
+                if (!rk->rk_conf.ssl.engine) {
```

Comment:
with the property .sdef set to "dynamic" this if is not necessary.

Reply:
it is required. 
flow is we MUST need to load engine as "dynamic" first.
subsequently we MUST reuse the engine by specified id